### PR TITLE
EZP-24872: Use a REST view to load the locations of a content

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-24872

## Description
Using a rest view instead of fetching locations one by one. This will improve the performances as less REST request will be needed.

## Tests
Manual tests on locations tab and update of unit tests

### CODE REVIEW

**Comment by yannickroger** [comment]
Ready for reviews @dpobel @mhyndle @StephaneDiot

File: Resources/public/js/models/ez-contentmodel.js [hunk 1]
```diff
@@ -232,45 +232,69 @@ YUI.add('ez-contentmodel', function (Y) {
...
+         * Loads locations of the content using a view. The result is the REST API view containing
+         * locations from the path.
+         *
+         * @method _loadLocationsView
+         * @protected
+         * @param {Object} options
+         * @param {Object} options.api the JS REST client instance
+         * @param {Function} callback
+         */
+        _loadLocationsView: function (options, callback) {
```
Comment (dpobel) [inline Resources/public/js/models/ez-contentmodel.js]: the name is a bit misleading actually I even wonder if this protected method is necessary, I guess you could put everything in `loadLocations`

File: Resources/public/js/models/ez-contentmodel.js [hunk 2]
```diff
@@ -232,45 +232,69 @@ YUI.add('ez-contentmodel', function (Y) {
...
-            var api = options.api,
-                currentLocation = options.location;
+            var locations = [],
+                that = this;
 
-            api.getContentService().loadLocations(this.get('id'), function (error, response) {
-                var tasks = new Y.Parallel(),
-                    loadError = false,
-                    locations = [];
+            this._loadLocationsView(options, function (error, response) {
```
Comment (dpobel) [inline Resources/public/js/models/ez-contentmodel.js]: if you keep this method, please bind the callback to `this` and remove the ugly `that = this`

**Comment by dpobel** [comment]
besides the naming/that nitpick, +1

**Comment by StephaneDiot** [comment]
+1

**Comment by yannickroger** [comment]
I took care of the feedback by @dpobel 
Ready to merge ?

**Comment by mhyndle** [comment]
+1

**Pull request merged**